internal/pkg/manifest/env_test.go

```diff
@@ -0,0 +1,243 @@
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			var env Environment
+			err := yaml.Unmarshal([]byte(tc.inContent), &env)
+			require.NoError(t, err)
+			require.Equal(t, tc.wantedStruct, env)
+		})
+	}
+}
+
+func TestEnvironmentVPCConfig_ImportedVPC(t *testing.T) {
+	testCases := map[string]struct {
+		inVPCConfig environmentVPCConfig
+		wanted      *template.ImportVPC
+	}{
+		"vpc not imported": {},
+		"only public subnets imported": {
+			inVPCConfig: environmentVPCConfig{
+				ID: aws.String("vpc-1234"),
+				Subnets: subnetsConfiguration{
+					Public: []subnetConfiguration{
+						{
+							SubnetID: aws.String("subnet-123"),
+						},
+						{
+							SubnetID: aws.String("subnet-456"),
+						},
+					},
+				},
+			},
+			wanted: &template.ImportVPC{
+				ID:              "vpc-1234",
+				PublicSubnetIDs: []string{"subnet-123", "subnet-456"},
+			},
+		},
+		"only private subnets imported": {
+			inVPCConfig: environmentVPCConfig{
+				ID: aws.String("vpc-1234"),
+				Subnets: subnetsConfiguration{
+					Private: []subnetConfiguration{
+						{
+							SubnetID: aws.String("subnet-123"),
+						},
+						{
+							SubnetID: aws.String("subnet-456"),
+						},
+					},
+				},
+			},
+			wanted: &template.ImportVPC{
+				ID:               "vpc-1234",
+				PrivateSubnetIDs: []string{"subnet-123", "subnet-456"},
+			},
+		},
+		"both subnets imported": {
+			inVPCConfig: environmentVPCConfig{
+				ID: aws.String("vpc-1234"),
+				Subnets: subnetsConfiguration{
+					Public: []subnetConfiguration{
+						{
+							SubnetID: aws.String("subnet-123"),
+						},
+						{
+							SubnetID: aws.String("subnet-456"),
+						},
+					},
+					Private: []subnetConfiguration{
+						{
+							SubnetID: aws.String("subnet-789"),
+						},
+						{
+							SubnetID: aws.String("subnet-012"),
+						},
+					},
+				},
+			},
+			wanted: &template.ImportVPC{
+				ID:               "vpc-1234",
+				PublicSubnetIDs:  []string{"subnet-123", "subnet-456"},
+				PrivateSubnetIDs: []string{"subnet-789", "subnet-012"},
+			},
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			got := tc.inVPCConfig.ImportedVPC()
+			require.Equal(t, tc.wanted, got)
+		})
+	}
+}
+
+func TestEnvironmentVPCConfig_ManagedVPC(t *testing.T) {
+	var (
+		mockVPCCIDR            = IPNet("10.0.0.0/16")
+		mockPublicSubnet1CIDR  = IPNet("10.0.0.0/24")
+		mockPublicSubnet2CIDR  = IPNet("10.0.1.0/24")
+		mockPublicSubnet3CIDR  = IPNet("10.0.2.0/24")
+		mockPrivateSubnet1CIDR = IPNet("10.0.3.0/24")
+		mockPrivateSubnet2CIDR = IPNet("10.0.4.0/24")
+		mockPrivateSubnet3CIDR = IPNet("10.0.5.0/24")
+	)
+	testCases := map[string]struct {
+		inVPCConfig environmentVPCConfig
+		wanted      *template.ManagedVPC
+	}{
+		"default vpc without custom configuration": {},
+		"with imported vpc": {
+			inVPCConfig: environmentVPCConfig{
+				ID: aws.String("vpc-1234"),
+			},
+		},
+		"with custom configuration to managed vpc": {
```

Comment:
nit: maybe rename this test case to be a bit more verbose putting the emphasis on what it's catching:
```suggestion
		"ensure custom configuration is sorted by AZ": {
```